python/text-to-speech/test_main.py

```diff
@@ -0,0 +1,309 @@
+            # Raise Exception.
+            mock_synthesize_speech.side_effect = Exception
+        # Assert the raise.
+        self.assertRaises(Exception, instance.speech, "hello", "en-EN")
+        self.assertRaises(Exception, instance.speech, "hello", None)
+
+    def test_speech_no_text(self) -> None:
+        """Test text-content for speech method."""
+        instance = self.get_google_instance(
+            "<YOUR_API_KEY>",
+            "<YOUR_PROJECT_ID>",
+        )
+        # Set mock.
+        with patch.object(texttospeech, "TextToSpeechClient") as mock_client:
+            # Raise Exception.
+            mock_client.side_effect = Exception
+        # Assert the raise.
+        self.assertRaises(Exception, instance.speech, None, "en-US")
+
+
+class AzureTest(unittest.TestCase):
+    """Azure API Test Cases."""
+    def get_azure_instance(self, key: str, project_id: str) -> main.Azure:
+        """Set Azure instance with request."""
+        req = MyRequest({
+            "payload": {
+                "provider": "azure",
+                "text": "hi",
+                "language": "en-US",
+            },
+            "variables": {
+                "API_KEY": key,
+                "REGION_KEY": project_id,
+            }
+        })
+        return main.Azure(req)
+
+    @parameterized.expand([
+        # Missing API_KEY.
+        (None, "123"),
+        # Missing REGION_KEY.
+        ("123", None),
+        # Missing Both.
+        (None, None),
+    ])
+    def test_validate_request(self, key: str, project_id: str) -> None:
+        """Test 'validate_request' method when fields missing or invalid."""
+        self.assertRaises(ValueError, self.get_azure_instance, key, project_id)
+
+    def test_speech_happy(self) -> None:
+        """Test speech method for successful TextToSpeech synthesis."""
+        instance = self.get_azure_instance("123", "123")
+        # Mock the requests.post method used in get_token.
+        with patch("requests.post") as mock_post:
+            mock_response = MagicMock()
+            mock_response.text = "fake_access_token"
+            mock_post.return_value = mock_response
+            # Mock the requests.request method.
+            with patch("requests.request") as mock_request:
+                mock_response_request = MagicMock()
+                mock_response_request.content = DECODED_RESULT_AZURE
+                mock_request.return_value = mock_response_request
+                # Call the speech method.
+                audio_bytes = instance.speech("hi", "en-US")
+                # Assert the result.
+                self.assertEqual(audio_bytes, DECODED_RESULT_AZURE)
+
+    def test_speech_invalid_credential(self) -> None:
+        """Test credentials for speech method."""
+        instance = self.get_azure_instance("WRONG_API_KEY", "WRONG_PROJECT_ID")
+        # Mock the requests.post method used in get_token.
```

Comment:
Try to avoid the temptation to write comments which says the same thing as the code.

Comments are great for explaining _why_ you are doing something or for explaining complicated code. For simple code, though, don't add comments which just repeat the code.